sdk/tests/platform/test_az_platform.c

```diff
@@ -0,0 +1,274 @@
+  {
+    assert_int_equal(az_platform_get_random(&test_random_2), AZ_OK);
+    assert_true(test_random_2 > -1);
+    i++;
+  }
+  assert_true(test_random_2 != test_random_1);
+}
+
+static void test_az_platform_timer_single(void** state)
+{
+  (void)state;
+
+  az_platform_timer test_timer_handle;
+  int64_t test_clock_one = 0; // Time before timer start.
+  int64_t test_clock_two = 0; // Time after timer callback.
+  test_timer_callback_counter = 0;
+
+  assert_int_equal(az_platform_timer_create(
+        &test_timer_handle,
+        _test_az_platform_timer_callback_set_time,
+        &test_clock_two),
+        AZ_OK);
+  assert_int_equal(az_platform_clock_msec(&test_clock_one), AZ_OK);
+  assert_int_equal(az_platform_timer_start(
+        &test_timer_handle, TEST_MILLISECOND_TIME), AZ_OK);
+
+  // Sleep until callback is triggered
+  assert_int_equal(az_platform_sleep_msec(2 * TEST_MILLISECOND_TIME), AZ_OK);
+
+  assert_int_equal(test_timer_callback_counter, 1);
+  assert_int_not_equal(test_clock_one, 0);
+  assert_int_not_equal(test_clock_two, 0);
+  assert_true((test_clock_two - test_clock_one) >= TEST_MILLISECOND_TIME);
+
+  assert_int_equal(az_platform_timer_destroy(&test_timer_handle), AZ_OK);
+}
+
+static void test_az_platform_timer_double(void** state)
+{
+  (void) state;
+  az_platform_timer test_timer_handle_1;
+  az_platform_timer test_timer_handle_2;
+  int64_t test_clock_one = 0;
+  int64_t test_clock_two = 0;
+  test_timer_callback_counter = 0;
+
+  assert_int_equal(az_platform_timer_create(
+        &test_timer_handle_1,
+        _test_az_platform_timer_callback_set_time,
+        &test_clock_one),
+        AZ_OK);
+  assert_int_equal(az_platform_timer_create(
+        &test_timer_handle_2,
+        _test_az_platform_timer_callback_set_time,
+        &test_clock_two),
+        AZ_OK);
+
+  assert_int_equal(az_platform_timer_start(
+        &test_timer_handle_1, TEST_MILLISECOND_TIME), AZ_OK);
+  assert_int_equal(az_platform_timer_start(
+        &test_timer_handle_2, 2 * TEST_MILLISECOND_TIME), AZ_OK);
+
+  assert_int_equal(az_platform_sleep_msec(3 * TEST_MILLISECOND_TIME), AZ_OK);
+
+  assert_int_equal(test_timer_callback_counter, 2);
+  assert_int_not_equal(test_clock_one, 0);
+  assert_int_not_equal(test_clock_two, 0);
+  assert_true(test_clock_two > test_clock_one);
+
+  assert_int_equal(az_platform_timer_destroy(&test_timer_handle_1), AZ_OK);
+  assert_int_equal(az_platform_timer_destroy(&test_timer_handle_2), AZ_OK);
+}
+
+static void test_az_platform_mutex(void** state)
+{
+  (void)state;
+
+  az_platform_mutex test_mutex_handle;
+
+  assert_int_equal(az_platform_mutex_init(&test_mutex_handle), AZ_OK);
+  assert_int_equal(az_platform_mutex_acquire(&test_mutex_handle), AZ_OK);
+  assert_int_equal(az_platform_mutex_release(&test_mutex_handle), AZ_OK);
+  assert_int_equal(az_platform_mutex_destroy(&test_mutex_handle), AZ_OK);
+}
+
+static void test_az_platform_mutex_double(void** state)
```

Comment:
Please add a new test to verify recursive (reentrant) mutex aquire (i.e. 2 or more aquires in the same thread should work - number of releases must still match number of acquires).